### PR TITLE
[Flang, Clang] Enable and test 'rdynamic' flag

### PR DESCRIPTION
**Pull request by Radu2k**
Enable and test 'rdynamic' flag

### CODE REVIEW

File: flang/test/Driver/rdynamic-check.f90
Comment (banach-space) [inline flang/test/Driver/rdynamic-check.f90:1]: No need for a dedicated test file, please use https://github.com/llvm/llvm-project/blob/main/flang/test/Driver/dynamic-linker.f90.

File: flang/test/Driver/rdynamic-check.f90 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+! Verify that rdynamic flag adds -export-dynamic flag and passes it on to the linker.
+
+! RUN: %flang -### --target=x86_64-linux-gnu -rdynamic %s 2>&1 | FileCheck --check-prefixes=GNU-LINKER-OPTIONS %s
+! RUN: %flang -### --target=aarch64-linux-none -rdynamic %s 2>&1 | FileCheck --check-prefixes=AARCH-LINKER-OPTIONS %s
```
Comment (banach-space) [inline flang/test/Driver/rdynamic-check.f90:4]: Both these run lines will use the GNU ToolChain within the driver. It's fine to delete one.

File: flang/test/Driver/dynamic-linker.f90 [hunk 2]
```diff
@@ -7,13 +7,17 @@
 ! RUN: %flang -### --target=x86_64-windows-msvc -rpath /path/to/dir -shared \
 ! RUN:     -static %s 2>&1 | FileCheck \
 ! RUN:     --check-prefixes=MSVC-LINKER-OPTIONS %s
+! RUN: %flang -### --target=aarch64-linux-none -rdynamic %s 2>&1 | FileCheck --check-prefixes=AARCH-LINKER-OPTIONS %s
```
Comment (banach-space) [inline flang/test/Driver/dynamic-linker.f90:10]: [nit] This run line is specifically for `-rdynamic` and the prefix should reflect that. Perhaps `RDYNAMIC`?

Comment (Radu2k) [reply flang/test/Driver/dynamic-linker.f90:10]: Sounds reasonable, will update soon. Thanks!

**Review by banach-space — APPROVED** [review]
LGTM, thanks!

**Pull request merged**